R/ResampleInstance.R

```diff
@@ -61,7 +61,26 @@ makeResampleInstance = function(desc, task, size, ...) {
   if (length(blocking) && desc$stratify)
     stop("Blocking can currently not be mixed with stratification in resampling!")
 
-  if (length(blocking)) {
+  # 'fixed' only exists by default for 'CV' -> is.null(desc$grouping)
+  # only use this way of blocking of 'fixed = FALSE' -> is.null(desc$grouping)
+  if(is.null(desc$fixed)) {
+    fixed = FALSE
+  } else {
+    if(desc$fixed == FALSE) {
+      fixed = FALSE
+      ### check if blocking should be used or not
+      # set to FALSE for all rdesc that do not have the arg
+      if(is.null(desc$blocking.cv)) {
+        blocking.cv = FALSE
+      } else {
+        blocking.cv = desc$blocking.cv
+      }
+    } else {
+      fixed = TRUE
+    }
+  }
```

Comment:
With default values for `fixed` and `blocking.cv` this could be simply `fixed = desc$fixed; blocking.cv = desc$blocking.cv`.

Reply:
Yes this simplifies things. I've set defaults in `makeResampleDesc()`.